gateway/ln-gateway/tests/integration_tests.rs

```diff
@@ -1393,6 +1400,71 @@ async fn test_gateway_can_leave_connected_federations() -> anyhow::Result<()> {
                 vec![3, 4]
             );
 
+            // send sats to first federation
+            info!("Sending sats to federation {id1}.");
+            send_msats_to_gateway(&gateway, id1, 5_000).await;
+            let fed1_balance = rpc.get_balance(BalancePayload {federation_id: id1}).await.expect("Could not query balance for federation {id1}.");
+            assert_ne!(fed1_balance.msats, 0u64);
+
+            // attempt to leave first federation while having a balance, unforced
+            rpc
+                .leave_federation(LeaveFedPayload {
+                    federation_id: id1,
+                    force_leave: false,
+                })
+                .await
+                .expect_err("Should have errored, gateway is trying to leave with a non-zero balance without the --forced flag.");
+
+            info!("attempt to leave the first federation while having a balance, forced");
+            rpc
+                .leave_federation(LeaveFedPayload {
+                    federation_id: id1,
+                    force_leave: true,
+                })
+                .await.expect("Should have been able to leave federation even though the gateway has a balance since the forced flag was passed.");
+
+            // send sats to second federation
+            send_msats_to_gateway(&gateway, id2, 5_000).await;
+            let fed2_balance = rpc.get_balance(BalancePayload {federation_id: id2}).await.expect("Could not query balance for federation {id2}.");
+            assert_ne!(fed2_balance.msats, 0u64);
+
+            // attempt to leave second federation while having a balance, unforced
+            rpc
+                .leave_federation(LeaveFedPayload {
+                    federation_id: id2,
+                    force_leave: false,
+                })
+                .await
+                .expect_err("Should have errored, gateway is trying to leave with a non-zero balance without the --forced flag.");
+
+            // attempt to leave the second federation while having a balance, forced
+            rpc
+                .leave_federation(LeaveFedPayload {
+                    federation_id: id2,
+                    force_leave: true,
+                })
+                .await.expect("Should have been able to leave federation even though the gateway has a balance since the forced flag was passed.");
+
+            rpc
+                .connect_federation(ConnectFedPayload {
+                    invite_code: invite1.to_string(),
+                })
+                .await
+                .unwrap();
+            rpc
+                .connect_federation(ConnectFedPayload {
+                    invite_code: invite2.to_string(),
+                })
+                .await
+                .unwrap();
+
+            let info = rpc.get_info().await.unwrap();
+            assert_eq!(info.federations.len(), 2);
+            assert_eq!(
+                info.channels.unwrap().keys().cloned().collect::<Vec<u64>>(),
```

Comment:
Nit: we're trying to reduce the number of `unwrap` in the code, lets change this to an `expect`

Reply:
I used `.expect_or_else` in some places because clippy didn't like putting `format` inside the `expect` modifier.